pkg/controller/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -0,0 +1,188 @@
+package ocsinitialization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ocsv1alpha1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_ocsinitialization")
+
+// watchNamespace is the namespace the operator is watching.
+var watchNamespace string
+
+const wrongNamespacedName = "Ignoring this resource. Only one should exist, and this one has the wrong name and/or namespace."
+
+// InitNamespacedName returns a NamespacedName for the singleton instance that
+// should exist.
+func InitNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name:      "ocsinit",
+		Namespace: watchNamespace,
+	}
+}
+
+// Add creates a new OCSInitialization Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileOCSInitialization{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// set the watchNamespace so we know where to create the OCSInitialization resource
+	ns, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return err
+	}
+	watchNamespace = ns
+
+	// Create a new controller
+	c, err := controller.New("ocsinitialization-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource OCSInitialization
+	return c.Watch(&source.Kind{Type: &ocsv1alpha1.OCSInitialization{}}, &handler.EnqueueRequestForObject{})
```

Comment:
Should this controller not also watch StorageClass resources, to reconcile actual against requested?